bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/model/InverterData.java

```diff
@@ -32,61 +31,117 @@ public interface InverterData extends RawDataBean {
 
     InverterType getInverterType();
 
-    short getInverterVoltage();
+    @Nullable
+    String getRawData();
 
-    short getInverterCurrent();
+    default double getPV1Voltage() {
+        return Integer.MIN_VALUE;
```

Comment:
Should it be:
```suggestion
        return Double.MIN_VALUE;
```

?

Can you point me to where this value is consumed? It should probably result in the channel being updated to `UNDEF`.

Reply:
These values are supposed to be overriden by the implementation of the interface but essentially decided to have default value if someone does not keep the Integer.MIN_VALUE, as double min value is too high and ugly. I think the conversion from integer to double is pretty safe in this case. 

> Can you point me to where this value is consumed? 

The usage is in the handler in the update channels section where we basically apply these values on the supported channels.

Reply:
Also I'm not sure if max/min double can be fit in the openhab numbered values for the channels...

Reply:
> Also I'm not sure if max/min double can be fit in the openhab numbered values for the channels...

Channels should not be updated with these values, they should be translated to `UnDefType.UNDEF`. You could consider boxing them to it would be possible to return `null` or [Optional.empty()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Optional.html#empty()).

Reply:
This is only a final resort value. In the regular use-case there should be
no channels that are not supported and do not return a value.
The final resort is if no one implements a channel but states in some
parser implementation(future, not current) that the channel is supported.

Maybe easiest would be because I use a private method to update the
channels, that first checks if the channel is supported, to only update it
if it's supported and value is > Short.MIN_VALUE (and make all default
values return Short.MIN_VALUE, including the Integer.MIN_VALUE ones). What
do you think?
If I have to refactor the interfaces to return optional and do
optional.orelse I'm not quite sure that the code will become more readable
or better...

Reply:
> Maybe easiest would be because I use a private method to update the
> channels, that first checks if the channel is supported, to only update it
> if it's supported and value is > Short.MIN_VALUE (and make all default
> values return Short.MIN_VALUE, including the Integer.MIN_VALUE ones). What
> do you think?

Sure, that would also work. I think one distinction is needed though:
- If not supported: Skip update.
- If supported, but `MIN_VALUE`: Update with `UNDEF`. Or - if this is to be considered a bug in the binding, you might instead log a warning instructing the user to report the bug.

Reply:
I did both now but had to do a refactoring that I was thinking about before the PR. Was a bit hesitant to do it. Tested the warning and the refactored code - works on my inverter - I've implemented as you suggested. If you decide to merge it, please hold on so I can get confirmation from the community that it also works on the new model.

Basically now I removed all single-three phase relevant data and treat all models the same way but rely that the parser "knows" which channel is supported and implemented. With this generic approach we can remove unnecessary classes and casts which are no longer required. Now the code is much more clean imho. 

Sorry for the late minute refactoring but I believe it's for the good of the code flow and the overall readability and extensibility.